openhands/runtime/impl/docker/docker_runtime.py

```diff
@@ -319,18 +319,23 @@ def _attach_to_container(self):
         self.container = self.docker_client.containers.get(self.container_name)
         if self.container.status == 'exited':
             self.container.start()
+
         config = self.container.attrs['Config']
         for env_var in config['Env']:
             if env_var.startswith('port='):
                 self._host_port = int(env_var.split('port=')[1])
                 self._container_port = self._host_port
             elif env_var.startswith('VSCODE_PORT='):
                 self._vscode_port = int(env_var.split('VSCODE_PORT=')[1])
+
         self._app_ports = []
-        for exposed_port in config['ExposedPorts'].keys():
-            exposed_port = int(exposed_port.split('/tcp')[0])
-            if exposed_port != self._host_port and exposed_port != self._vscode_port:
-                self._app_ports.append(exposed_port)
+        exposed_ports = config.get('ExposedPorts')
+        if exposed_ports:
+            for exposed_port in exposed_ports.keys():
+                exposed_port = int(exposed_port.split('/tcp')[0])
+                if exposed_port != self._host_port and exposed_port != self._vscode_port:
```

Comment:
Huh I thought we had a better lookup here...the app ports do have specific ranges.